examples/loki/nurf_data.py

```diff
@@ -107,115 +107,163 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
     
-        # create the various datasets for UV
-        uv_inttime = grp_uv.create_dataset('uv_integration_time',
+        uv_inttime_data=uv_inttime.create_dataset('uv_integration_time',
                                            data=data['UV_IntegrationTime'],
                                            shape=data[
                                                'UV_IntegrationTime'].shape,
                                            dtype=np.int32)
-        uv_inttime.attrs['long_name'] = 'uv_integration_time'
-        uv_inttime.attrs['units'] = 's'  # TODO: unit to be verified
+        uv_inttime_data.attrs['long_name'] = 'uv_integration_time'
+        uv_inttime_data.attrs['units'] = 's'  # TODO: unit to be verified
 
-        uv_bkg = grp_uv.create_dataset('uv_background',
+        # subgroup for uv_background
+        uv_bkg=grp_uv.create_group("uv_background")
+        uv_bkg.attrs["NX_class"] = 'NXdetector'
+
+        uv_bkg_data = uv_bkg.create_dataset('data',
                                        data=data['UV_background'],
                                        shape=data['UV_background'].shape,
                                        dtype=np.float32)
-        uv_bkg.attrs['long name'] = 'uv_background'
-
-        uv_int0 = grp_uv.create_dataset('uv_intensity0',
-                                        data=data['UV_intensity0'],
-                                        shape=data['UV_intensity0'].shape,
-                                        dtype=np.float32)
-        uv_int0.attrs['long name'] = 'uv_intensity0'
-
-        uv_spectra = grp_uv.create_dataset('uv_spectra',
-                                           data=data['UV_spectra'],
-                                           shape=data['UV_spectra'].shape,
-                                           dtype=np.float32)
-        uv_spectra.attrs['long name'] = 'uv_spectra'
+        uv_bkg_data.attrs['long name'] = 'uv_background'
+        uv_bkg_data.attrs['data'] = 'a.u.'
+
+        # subgroup for uv_intensity0
+        uv_int0=grp_uv.create_group("uv_intensity0")
+        uv_int0.attrs["NX_class"] = 'NXdetector'
+        uv_int0_data = uv_int0.create_dataset('data', data=data['UV_intensity0'], 
+                                            shape=data['UV_intensity0'].shape, 
+                                            dtype=np.float32)
+        uv_int0_data.attrs['long name'] = 'uv_intensity0'
+        uv_int0_data.attrs['units'] = 'a.u.'
 
-        uv_wavelength = grp_uv.create_dataset('uv_wavelength',
+        
+        # subgroup for uv_wavelength
+        uv_wavelength=grp_uv.create_group("uv_wavelength")
+        uv_wavelength.attrs["NX_class"] = 'NXdetector'
+        
+        uv_wavelength_data = uv_wavelength.create_dataset('data',
                                               data=data['UV_wavelength'],
                                               shape=data['UV_wavelength'].shape,
                                               dtype=np.float32)
```

Comment:
As written above, this should be within the `NXdetector` for the dependent (measured) spectra. They should (each) define an `axes` attribute (on the `NXdetector`), and a `wavelength` dataset within that group.

If the wavelength is the same for all types of measured spectra, you can consider using links, if size is a concern (I suppose it is not, because `wavelength` will not change, i.e., is 1-D, whereas data is 2-D (at least)?).